### ISSUE TITLE
STUDY GROUPS: Feature Feedback

### ISSUE THREAD

**Issue by lpatmo** [codebuddies/codebuddies#563]
Would appreciate any feedback on the new study groups feature found now on http://staging.codebuddies.org.

Please leave your feedback as a comment here, or on Slack in the #codebuddies-meta channel.

INSTRUCTIONS:
1. Log in to http://staging.codebuddies.org
2. Create a new study group, edit the description, and add some links to it under the "Resources" tab.
3. Join an existing study group

QUESTIONS:
1. Do you know how to make another member of the group a moderator or admin so that they can also schedule hangouts for the study group?
2. What about the interface could be improved, or what additional features would you like to see for each study group?
3. Does the 24/7 hangout appear useful?

**Comment by lpatmo** [comment]
> 2. What about the interface could be improved, or what additional features would you like to see for each study group??
Some ideas:
- Add a number next to the (Links) tab so we know how many links are a part of the group
- Add a number next to the (Members) tab so we have a count of the total number of members in the group
- Allow members to subscribe to email notifications for the group whenever a new hangout is scheduled as part of the group (https://github.com/codebuddiesdotorg/codebuddies/issues/558)
- Maybe we could have a "Questions" tab where members can post questions related to the study group topic? And other folks could post answers?

> 3. Does the 24/7 hangout appear useful?
I can see myself using it whenever I'm studying material related to the study group. And it'll be a place where I can find spontaneous study partners.

**Comment by osanseviero** [comment]
I would change the "links" tab to "resources". Sounds fancier.

**Comment by tonypnode** [comment]
Looks great!

When trying to add a new study group I didn't put in a tag line. The only indication of the error was a flash message that said "Study Group Name". You could put a more detailed error notifying the user that a tag line is required. Or place a "required" indicator next to the Tag field.

**Comment by tonypnode** [comment]
Also, not sure if it was because I was the only one in the hangout, but while entering text into the jtsi chat... nothing happened when I pressed enter. The message did not appear in the chat log.

Idea - (Not sure if it's already there or not) Audible/visual notification of member joining Jitsi hangout
 - If I am the only person in the hangout, I will be working on my project or looking at different pages. It would be nice to get a notification that a new study buddy has joined

**Comment by lpatmo** [comment]
Thanks @osanseviero and @tonypnode! :) I've filed all of our suggestions as issues:

https://github.com/codebuddiesdotorg/codebuddies/issues/564 - "Resources" instead of links
https://github.com/codebuddiesdotorg/codebuddies/issues/565 - more descriptive error message for required tagline
https://github.com/codebuddiesdotorg/codebuddies/issues/566 - add number counts next to "members" and "resources"

@tonypnode: What browser were you using, out of curiosity? I wonder if it could be related to the chat message problem you were seeing.

Re: audible/visual notification of another member joining a hangout -- Jitsi will automatically play a sound when a new member joins, the number count of members joined will increase on the left bar, a second icon will also appear. 

I believe the Jitsi API also lets us create our own event notifications when participants join or leave (https://github.com/jitsi/jitsi-meet/blob/master/doc/api.md), although I'm not exactly sure what that should look like that'll enhance what Jitsi already does. For example, should we display an alert on the page? Would that be too distracting? Normally new participants will type 'hi' in the chat or have their microphone on and say hi, which would alert existing users that they're there.

Thanks everyone! Please keep the feedback coming.

**Comment by tonypnode** [comment]
using, Safari 10.1.2 on MacOS 10.12.6. 

Re:Re visual notifs - that makes sense. Sounds like the feature is already there.

**Comment by lpatmo** [comment]
@tonypnode Got it; did you see a warning about using Chrome or Firefox? (See screenshot below.) I wonder how we can edit that message so that it's clearer that using Jitsi in other browsers like Safari might result in problems.

![screen shot 2017-09-10 at 2 05 06 pm](https://user-images.githubusercontent.com/4512699/30253064-6e181602-9632-11e7-8ae3-701d01365cff.jpg)

By the way, I've merged in some fixes/updates to staging, so you should see the improvements at https://staging.codebuddies.org/study-group/obey-the-testing-goat-tdd-with-python-and-django/69c5wndowNYGWZyFy now. Please let me know if you see any issues or have other ideas about how study groups could be improved! Thanks!

**Comment by osanseviero** [comment]
I'm not able to create a study group from the my study groups page. The button does not seem to respond. I don't have this problem when clicking in + > new study group.

**Comment by osanseviero** [comment]
Some additional feedback.
<img width="1093" alt="screen shot 2017-09-10 at 14 18 39" src="https://user-images.githubusercontent.com/7246357/30253084-00091552-9633-11e7-85e2-77196fe3b853.png">

For good user experience, try to contain things were they belong. Here the website is giving feedback on the form outside the modal, which seems a bit messy. This could potentially cause a11y issues as well.

**Comment by lpatmo** [comment]
@osanseviero You were on https://staging.codebuddies.org/study-groups, and clicking on the "+" icon in the nav to create the study group? What browser are you using? There were some fixes deployed to the staging site a few minutes ago, which might've interfered with the website at that point.

**Comment by osanseviero** [comment]
@lpatmo  Nope. In My Study Groups page, when you haven´t joined a study group yet, there was a button to join a study group and one to find a study group. Neither of these buttons worked.

**Comment by osanseviero** [comment]
Sorry for multiple comment, I'm trying to provide feedback on issues separately. How can I exit a study group?

**Comment by lpatmo** [comment]
No worries, thanks for the feedback! To leave a study group, there is a button that you can click on:

![screen shot 2017-09-10 at 2 24 33 pm](https://user-images.githubusercontent.com/4512699/30253144-ed52d73a-9633-11e7-9b99-9eae6a6d76aa.jpg)

**Comment by tonypnode** [comment]
@lpatmo . I didn't see the warning haha.. sorry. Chat works just fine in FF.

**Comment by osanseviero** [comment]
Oh, I'm blind!

Ok, so I was worried about this. As a tip, maybe pagination could work for the activity, and give it in reverse order.

<img width="459" alt="screen shot 2017-09-10 at 14 28 22" src="https://user-images.githubusercontent.com/7246357/30253165-686c2836-9634-11e7-96b8-8076ef964a39.png">

**Comment by osanseviero** [comment]
Also, I'm really curious about the 24/7 hangouts. It seems really interesting and with high potential, but I do see cases where the organizers simply don't want that feature in their study group. Maybe there should be an option to add or remove the 24/7 hangouts? 

What user scenarios do you see for the 24/7?

**Comment by osanseviero** [comment]
+bug

I already left a study group but it still appears in my study groups
<img width="1271" alt="screen shot 2017-09-10 at 14 39 44" src="https://user-images.githubusercontent.com/7246357/30253244-ff1db870-9635-11e7-9482-79d2be9a7971.png">
<img width="976" alt="screen shot 2017-09-10 at 14 39 58" src="https://user-images.githubusercontent.com/7246357/30253243-ff1a8c36-9635-11e7-923b-9f27745886ea.png">

**Comment by lpatmo** [comment]
Man, I wish Github had "nested replies" as a feature on issue threads. Thanks guys for all this great feedback -- and btw, if you want to create a new issue starting with "STUDY GROUPS" in the subject line instead of posting it as a comment here, feel free to do so! Otherwise, I'll continue to copy/paste comments here into new issues.

> What user scenarios do you see for the 24/7?
Good question! :D A couple of cases:
- It takes a bit of organizing work to schedule a hangout -- finding others who can meet at X time, filling out the time on the create hangout modal, setting a title, and writing a description. Hence why we don't see a lot of hangouts scheduled.

- Person A and B might be studying the same material at the same time (e.g. the same book, or interested and finding a pair programming partner to do some CodeWars exercises with) but at the moment this requires that they attempt to find study partners by typing a question on Slack. And oftentimes people are too lazy to do this, or find the proposal several hours too late.

- Study groups help organize people who are interested in studying the same specific topic, and saves people the step of having to ask around for study partners (or organize the time and details of a hangout) because they can simply log in to the 24/7 hangout when they feel like studying, and potentially find other study partners interested in studying *that* particular thing immediately.

- An additional bonus of Jitsi hangouts: the chat in the chatbox is archived FOREVER, so even if you exit the 24/7 hangout you can still see the chat logs from when you were away from other members of the study group.

I'm curious: why do you think organizers might want to turn off the 24/7 hangout? This could be its own issue thread too.

**Comment by osanseviero** [comment]
Ok, I understand better the feature now.

As I see it, if there's a study group like "JavaScript Learners", the 24/7 hg makes sense. If there´s a group like "Weekly Pythonic Review", or something that has a "fixed" schedule (e.g. weekend hackathon), they might not see much use for the 24/7 hg. Just my two cents. You're the one that has much more experience in the platform and the use of the hangouts.

**Comment by lpatmo** [comment]
I think even if a group like "weekly python review" or "weekend civic hackathon" has a habit of meeting formally for a hangout only once a week, group members might still end up doing "homework" or spend some time related to the project beyond the weekly meeting time -- and so might find having a 24/7 space to chat or jump on for a spontaneous screenshare useful.

**Comment by arturolei** [comment]
There does not readily appear to be a way to alter or change Study Group name; this feature might be needed in the future.  Also, unclear is how to leave or rejoin a study group. 

One other suggestion: when you click on a study group page, maybe the 24/7 hangout should not be the first tab open. The first tab open should be one that contains general information. In fact, it's hard to see the descripton of the group. Perhaps, there should be a splash/landing page that is not a 24/7 hangout.

Also, the above instructions state "Links" tab but at least for me it was "Resources" tabs.

**Comment by osanseviero** [comment]
I agree with @arturolei that it isn´t clear how to leave a group. I know that the button is right there, but maybe changing the color to red or putting it in another place might be better.

**Comment by osanseviero** [comment]
Sorry, I just saw that there's already an issue for that. I will create some issues for other bugs.

I see your point about the 24/7 hg, @lpatmo. Sounds great. What do you think about the possibility of adding tags to your study groups? Tags like JavaScript, Rails, etc. Even if right now they just work as tags, there could be some interesting possibilities for searching and for connecting with Slack (e.g. announce in the Python channel when there is something related to a study group of python).

**Comment by abdulhannanali** [comment]
@osanseviero That would lead to too much noise IMHO though, if we have mail notifications or push notifications they should be enough for alerting the users, from my experience with CodeBuddies the public  rooms on slack are very much for the human discussions. I am in support of having an integration to a Slack Webhook so people can do all type of cool stuff with them and even configure it for their private or custom channels, but for public channels such as JavaScript, Ruby etc. will lead to too much unnecessary noise and will probably undermine the human conversations happening here. Also, private slack has a nice 10,000 message limit there.

**Comment by abdulhannanali** [comment]
### Study Groups Feature feedback
I gave study groups feature a try on [Codebuddies Staging](https://staging.codebuddies.com) and I am excited about it's release and looking forward to try it out in the production and re engage with the wonderful community of code buddies. Study groups  provides an instant way to connect with the people and it can make the new members of the codebuddies community productive from the day one since they can just join groups that are of their interest and start interacting with the community without ever leaving the site. 

There was a usual flow that I think people had to follow before, Schedule a Hangout -> Go to Slack -> Spread about the hangout and committing to these hangouts require extraordinary willpower and commitment for both the organizer and other participants in the hangout which is not always possible because of the Timezones and other differences. Different interaction features such as Resources, Learnings and 24/7 Hangout can be very useful. 

#### How do I perceive StudyGroups?

Study groups for me is an initiative to have more organized discussion and learning on the codebuddies platform out of slack where people intend to come with the willingness to learn with others. I see it as an idea to let people learn and interact without going to slack if they don't want to, obviously they can go there but this feature gives much more organization to all that is being learnt which was lost because of the Synchronous Way in which the Slack works and truly form new type of communities around different topics.

### StudyGroups should act like groups, more about 24/7 hangout

24/7 Hangout is a brilliant and new idea and that's why I am confused about it, many groups are going to be small at first and probably people are not going to be online in groups 24/7 so it provides an unorganization on the level of groups and we again have people who are missing on stuff, features such as Learnings, Chat Archives and Resources help in getting up to date with what people are upto though, but we shouldn't expect people to read Jitsi Archives, that's just plain hard to do, does not seem any better than slack. 

Meanwhile All Hangouts provides what I expected, a better way to manage the hangouts related to a specific group, and other features that assist are Learnings, Resources and hopefully questions. 

### Features I think will be great

@lpatmo  posted about consideration of the "Questions" feature within the application, this feature can help in improving the interaction within the CodeBuddies platform for the study groups while providing access to all the knowledge from within here. We can help people as we can have a centralized database of questions people ask repeatedly, and given that many people in StudyGroups are going to be at same stages of knowledge, an answer can benefit many. 

### Answers to questions @lpatmo asked in the issue description

It was pretty trivial to find, as expected I went to the Members tab, clicked on Modify and woosh, but for the ease of access you can add Make Admin on the Members without clicking Modify to make it easier and also provide a confirmation screen. 

Any improvements to the interface?

How about renaming Progress Log to Learnings again, I like the latter one more, Progress Log gives it a feel of something that it's automated and you go into them mostly when something's wrong, that would discourage the usage of this feature. 

If something does not make sense, @ me and I would love to discuss :)

**Comment by billglover** [comment]
Closing this issue as Study Groups feature appears to be live on the main site.
